tests/backup/backup_helper.go

```diff
@@ -1426,3 +1435,200 @@ func DeleteBackupAndWait(backupName string, ctx context.Context) error {
+
+		for _, job := range allJobs.Items {
+			if strings.Contains(job.Name, pxCentralPostInstallHookJobName) {
+				err = deleteJobAndWait(job)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	// Get storage class using for px-backup deployment
+	statefulSet, err := apps.Instance().GetStatefulSet(mongodbStatefulset, pxBackupNamespace)
+	if err != nil {
+		return err
+	}
+	pvcs, err := apps.Instance().GetPVCsForStatefulSet(statefulSet)
+	if err != nil {
+		return err
+	}
+	storageClassName := pvcs.Items[0].Spec.StorageClassName
+
+	// Get the tarball required for helm upgrade
+
+	cmd = fmt.Sprintf("curl -O  https://raw.githubusercontent.com/portworx/helm/%s/stable/px-central-%s.tgz", latestPxBackupHelmBranch, versionToUpgrade)
+	log.Infof("curl command to get tarball: %v ", cmd)
+	output, _, err := osutils.ExecShell(cmd)
+	if err != nil {
+		return fmt.Errorf("error downloading of tarball: %v", err)
+	}
+	log.Infof("Terminal output: %s", output)
+
+	// Execute helm upgrade using cmd
+
+	log.Infof("Upgrading Px-Backup version from %s to %s", currentBackupVersionString, versionToUpgrade)
+	cmd = fmt.Sprintf("helm upgrade px-central px-central-%s.tgz --namespace %s --create-namespace --version %s --set persistentStorage.enabled=true,persistentStorage.storageClassName=\"%s\",pxbackup.enabled=true",
+		versionToUpgrade, pxBackupNamespace, versionToUpgrade, *storageClassName)
+	log.Infof("helm command: %v ", cmd)
+	output, _, err = osutils.ExecShell(cmd)
+	if err != nil {
+		return fmt.Errorf("upgrade failed with error: %v", err)
+	}
+	log.Infof("Terminal output: %s", output)
+
+	// Wait for post install hook job to be completed
+	postInstallHookJobCompletedCheck := func() (interface{}, bool, error) {
+		job, err := batch.Instance().GetJob(pxCentralPostInstallHookJobName, pxBackupNamespace)
+		if err != nil {
+			return "", true, err
+		}
+		if job.Status.Succeeded > 0 {
+			log.Infof("Status of job %s after completion - "+
+				"\nactive count - %d"+
+				"\nsucceeded count - %d"+
+				"\nfailed count - %d\n", job.Name, job.Status.Active, job.Status.Succeeded, job.Status.Failed)
+			return "", false, nil
+		}
+		return "", true, fmt.Errorf("status of job %s not yet in desired state - "+
+			"\nactive count - %d"+
+			"\nsucceeded count - %d"+
+			"\nfailed count - %d\n", job.Name, job.Status.Active, job.Status.Succeeded, job.Status.Failed)
+	}
+	_, err = task.DoRetryWithTimeout(postInstallHookJobCompletedCheck, 10*time.Minute, 30*time.Second)
+	if err != nil {
+		return err
+	}
+
+	// Checking if all pods are running
+	allPods, err := core.Instance().GetPods(pxBackupNamespace, nil)
+	log.Infof("Listing all the pods in Px Backup Namespace [%s] - ", pxBackupNamespace)
+	for _, pod := range allPods.Items {
+		log.Infof(pod.GetName())
+	}
+	if err != nil {
+		return err
+	}
+	for _, pod := range allPods.Items {
+		log.Infof("Checking status for pod - %s", pod.GetName())
+		core.Instance().ValidatePod(&pod, 5*time.Minute, 30*time.Second)
+	}
```

Comment:
Did you check for error later because you're sure that _even if there is an error_, there will _always_ be a list of all pods to go through and that list is useful even if there's an error?
Shouldn't error check be done first?